README.md

```diff
@@ -5,3 +5,157 @@
+Now you can clone the project locally and get to work!
+
+    $ git clone https://github.com/<user>/<your_new_repo>.git
+
+### Remove frameworks you're not going to use
+If you know you're not going to use one or more of the optional gui/graphics 
+frameworks (fltk, gtkmm, imgui, etc.), you can remove them with `git rm`:
+
+    $ git rm -r src/<unnecessary_framework>
+
+## Dependencies
+
+### Necessary Dependencies
+1. A C++ compiler that supports C++17 (gcc 7+, clang 6+, Visual Studio 2019)
+2. [Conan](https://conan.io/) - it's recommended that you install using 
+[pip](https://pip.pypa.io/en/stable/) 
+3. [CMake](https://cmake.org/)
+
+### Optional Dependencies
+This project can be made to work with several optional GUI frameworks. 
+    
+If desired, you should install the following optional dependencies as 
+directed by their documentation, linked here:
+ 
+- [FLTK](https://www.fltk.org/doc-1.4/index.html)
+- [GTKMM](https://www.gtkmm.org/en/documentation.html)
+- [QT](https://doc.qt.io/)
+
+The following dependencies can be downloaded automatically by CMake and Conan. 
+All you need to do to install them is to turn on a CMake flag during 
+configuration.
+If you run into difficulty using them, please refer to their documentation,
+linked here:
+
+- [NANA](http://nanapro.org/en-us/documentation/)
+- [SDL](http://wiki.libsdl.org/FrontPage)
+- [IMGUI](https://github.com/ocornut/imgui/tree/master/docs): 
+  This framework depends on SFML, and if you are using Linux, you may need
+  to install several of SFML's dependencies using your package manager. See 
+  [the SFML build tutorial](https://www.sfml-dev.org/tutorials/2.5/compile-with-cmake.php)
+  for specifics.
+
+## Build Instructions
+
+### Make a build directory
+
+    $ mkdir build && cd build
+
+### Configure your build
+
+To configure the project and write makefiles, you could use `cmake` with a
+bunch of command line options. The easier option is to run cmake interactively,
+with the Cmake Curses Dialog Command Line tool:  
+
+    $ ccmake ..
+
+Once `ccmake` has finished setting up, press 'c' to configure the project, 
+press 'g' to generate, and 'q' to quit.
+
+### Build
+Once you have selected all the options you would like to use, you can build the 
+project:
+
+    $ cmake --build .   # build all targets
+
+### Build using an alternate compiler
+
+If you need to use a compiler other than the system default, you will need to 
+do two things:
+
+* Tell Conan which compiler to use, via the environment variables CC and CXX
+* Tell CMake which compiler to use, via the CMake variables CMAKE_C_COMPILER 
+  and CMAKE_CXX_COMPILER
+
+Be aware that CMake will detect which compiler was used to build each of the 
+Conan targets, and if it doesn't match CMAKE_C_COMPILER or CMAKE_CXX_COMPILER,
+the project will fail to build.
+
+To build using clang, you can use these commands:
+
+    $ CC=clang CXX=clang++ ccmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ..
```

Comment:
I thought `CMAKE_C_COMPILER` would default to `CC`

Reply:
Yes: https://cmake.org/cmake/help/v3.10/envvar/CC.html

Reply:
You're right. I think I meant to change this at one point but never got around to it. Thanks for catching this.